### PR TITLE
37661: A New Experience for Discovering, Installing, and Previewing Themes in the Customizer

### PR DESCRIPTION
**Pull request by westonruter**
See https://core.trac.wordpress.org/ticket/37661

### CODE REVIEW

File: src/wp-includes/class-wp-customize-manager.php [hunk 1]
```diff
@@ -1777,6 +1788,7 @@ public function get_nonces() {
 		$nonces = array(
 			'save' => wp_create_nonce( 'save-customize_' . $this->get_stylesheet() ),
 			'preview' => wp_create_nonce( 'preview-customize_' . $this->get_stylesheet() ),
+			'customize-themes' => wp_create_nonce( 'customize-themes' ),
```
Comment (westonruter) [inline src/wp-includes/class-wp-customize-manager.php]: This nonce should only be output if `current_user_can( 'switch_themes' )`. Also, maybe `switch-themes` is a better nonce action than `customize-themes`?

File: src/wp-includes/class-wp-customize-manager.php [hunk 2]
```diff
@@ -2341,6 +2391,142 @@ public function register_dynamic_settings() {
...
+					'homepage' => true,
+					'num_ratings' => true,
+					'tags' => true,
+				),
+			);
+
+			// Specialized handling for each query.
+			switch ( $_POST['theme_action'] ) {
+				case 'search':
+					$args['search'] = wp_unslash( $_POST['search'] );
```
Comment (westonruter) [inline src/wp-includes/class-wp-customize-manager.php]: Sanitization for this and the other vars in `$_POST`?

File: src/wp-includes/class-wp-customize-manager.php [hunk 3]
```diff
@@ -2458,10 +2457,11 @@ public function load_themes_ajax() {
 					$args['search'] = wp_unslash( $_POST['search'] );
 					break;
 				case 'favorites':
-					$args['user'] = wp_unslash(  $_POST['user'] );
+					$args['user'] = wp_unslash( $_POST['user'] );
+					break;
```
Comment (celloexpressions) [inline src/wp-includes/class-wp-customize-manager.php]: There was intentionally no `break` here - the `browse` arg is also required for the favorites API query.

**Comment by celloexpressions** [comment]
@westonruter I think I fixed most of your initial comments, let me know if you find any other issues.

File: src/wp-includes/customize/class-wp-customize-theme-control.php [hunk 4]
```diff
@@ -66,42 +66,59 @@ public function content_template() {
...
-				<span class="more-details" id="{{ data.theme.id }}-action"><?php _e( 'Live Preview' ); ?></span>
-			<# } #>
+			<span class="more-details theme-details" id="{{ data.section }}-{{ data.theme.id }}-action" aria-label="<?php echo $details_label; ?>"><?php _e( 'Theme Details' ); ?></span>
 
-			<div class="theme-author"><?php printf( __( 'By %s' ), '{{ data.theme.author }}' ); ?></div>
 
-			<# if ( data.theme.isActiveTheme ) { #>
-				<h3 class="theme-name" id="{{ data.theme.id }}-name">
+			<# if ( 'installed' === data.theme.type && data.theme.hasUpdate ) { #>
+				<div class="update-message notice inline notice-warning notice-alt" data-slug="{{ data.theme.id }}"><p><?php _e( 'New version available. <button class="button-link update-theme" type="button">Update now</button>' ); ?></p></div>
```
Comment (celloexpressions) [inline src/wp-includes/customize/class-wp-customize-theme-control.php]: Yes, I'll work on this now

File: src/wp-admin/js/customize-controls.js [hunk 5]
```diff
@@ -2168,12 +2156,9 @@
...
 
 			if ( expanded ) {
-				overlay.addClass( 'in-themes-panel' ).addClass( 'showing-themes' )
-				       .delay( 200 ).find( '.customize-themes-full-container' ).addClass( 'animate' );
-
-				if ( false === api.state( 'saved' ).get() ) {
-					panel.container.find( '.customize-themes-unsaved-changes' ).show();
-				}
+				overlay
+					.addClass( 'in-themes-panel' ).addClass( 'showing-themes' )
```
Comment (westonruter) [inline src/wp-admin/js/customize-controls.js]: Yes. Letting tab indentation logically group the jQuery methods.

**Comment by westonruter** [comment]
Committed to core in https://core.trac.wordpress.org/changeset/38813

**Pull request closed without merging**